dcos/http/transport/option.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2016 Mesosphere, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transport
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"time"
+)
+
+var (
+	// ErrInvalidCredentials is the error returned by NewRoundTripper if user used empty string for a credentials.
+	ErrInvalidCredentials = errors.New("uid, secret and loginEndpoit cannot be empty")
+
+	// ErrInvalidExpireDuration is the error returned by NewRoundTripper if the token expire duration is negative or
+	// zero value.
+	ErrInvalidExpireDuration = errors.New("token expire duration must be positive non zero value")
+)
+
+// OptionTransportFunc type sets optional configurations for the
+// DC/OS HTTP transport
+type OptionTransportFunc func(*dcosTransport) error
+
+// OptionRoundtripperFunc type sets options configurations for the
+// DC/OS HTTP roundtripper
+type OptionRoundtripperFunc func(*dcosRoundtripper) error
+
+func errorOnEmpty(arg string) error {
+	if len(arg) == 0 {
+		return errors.New("Must pass non-empty string to this option")
```

Comment:
can we extract the error and make it package global, similar to `ErrInvalidCredentials` and `ErrInvalidExpireDuration`